logistics_project/apps/malawi/warehouse/runner.py

```diff
@@ -501,34 +495,57 @@ def update_consumption_times(since):
             c.time_needing_data = c.time_with_data
             c.save()
 
+
 def update_historical_data():
     """
     If we don't have a record of this supply point being updated, run
     through all historical data and just fill in with zeros.
     """
-    start_date = ReportingRate.objects.order_by('date')[0].date
-    warehouse_classes = [
+    hsa_level_start_date = ReportingRate.objects.filter(base_level=BaseLevel.HSA).order_by('date')[0].date
```

Comment:
did you not hard code these values already elsewhere for simplicity & performance reasons?